src/lib/y2network/interface.rb

```diff
@@ -25,13 +25,39 @@ class Interface
     attr_accessor :description
     # @return [Symbol] Interface type
     attr_accessor :type
+    attr_reader :configured
+    attr_reader :hardware
+
+    # Shortcuts for accessing interfaces' ifcfg options
+    ["STARTMODE", "BOOTPROTO"].each do |ifcfg_option|
+      method_name = ifcfg_option.downcase
+
+      define_method method_name do
+        # when switching to new backend we need as much guards as possible
+        if !configured || config.nil? || config.empty?
+          raise "Trying to read configuration of an unconfigured interface"
+        end
+
+        config[ifcfg_option]
+      end
+    end
 
     # Constructor
     #
     # @param name [String] Interface name (e.g., "eth0")
     def initialize(name)
-      @name = name
       @description = ""
+      # @hardware and @name should not change during life of the object
+      @hardware = Hwinfo.new(name: name)
+
+      if !(name.nil? || name.empty?)
+        @name = name
+      elsif @hardware.nil?
+        # the interface has to be either configured (ifcfg) or known to hwinfo
+        raise "Attempting to create representation of nonexistent interface"
```

Comment:
please add to documentation of method that name can be empty or nil and what happen in such case.

Reply:
done